src/frontend/src/lib/canisters/signer.canister.ts

```diff
@@ -91,28 +97,65 @@ export class SignerCanister extends Canister<SignerService> {
 		return address;
 	};
 
-	signTransaction = ({ transaction }: { transaction: SignRequest }): Promise<string> => {
-		const { sign_transaction } = this.caller({
+	signTransaction = async ({
+		transaction
+	}: {
+		transaction: EthSignTransactionRequest;
+	}): Promise<string> => {
+		const { eth_sign_transaction } = this.caller({
 			certified: true
 		});
 
-		return sign_transaction(transaction);
+		const response = await eth_sign_transaction(transaction, [SIGNER_PAYMENT_TYPE]);
+
+		// If the response does not match the type signature, so has neither `Ok` nor `Err`,
+		// will typescript have thrown an error before this point?  Ditto for the other APIs.
+		// It seems safer to check for `Ok` in response, and always throw an error if it's not there.
+
+		if ('Ok' in response) {
```

Comment:
That's probably more a nitpick but, given that in existing functions we check for the `if Err` and then return the response, I would apply the same order here for consistency reasons.

```
if ("Err" in response)
   throw mapSigner...
   
const {Ok: {signature}} = response;
return signature
```

On the contrary, if it makes more sense to check for "Ok" first, then let's rework the existing functions in another PR.

WDYT?

Reply:
I don't know whether typescript throws an exception if, at runtime, the wrong data type is provided by an API.  If it does throw an exception, it doesn't matter which patetrn is used.  If it doesn't, checking for Ok handles wacky data.  So I would rather convert everything to the defensive version, but if you are 100% sure that typescript enforces types at runtime I can switch to the pattern used elsewhere.

Reply:
It looks as if there is no runtime check, so I'll convert the existing methods.  But I'll do that in a separate PR to avoid mixing topics.

Reply:
At build time, TypeScript prevents it indeed.
At runtime, Candid should throws (I think), if anything other than Ok or Err is provided.

That said, I think you make a good point anyway, so let's rather convert everything to defensive and move on with this PR.

Reply:
Thanks.  Yes, it would good to have a close look at the candid parser so that we know what integrity guarantees it provides.